tncc.py

```diff
@@ -564,13 +571,34 @@ def get_cookie(self, dspreauth=None, dssignin=None):
         post_data = ''.join([ '%s=%s;' % (k, v) for k, v in post_attrs.iteritems()])
         self.r = self.br.open('https://' + self.vpn_host + self.path + 'hc/tnchcupdate.cgi', post_data)
 
+        # set the inital host checker cookie to the first DSPREAUTH cookie
+        self.set_cookie('DSPREAUTH_HC', self.find_cookie('DSPREAUTH').value)
+
+        # parse the response to retrieve the periodic host checking interval
+        response = self.parse_response()
+
+        # try to get the interval from the response, if not fallback with except
+        try:
+            self.hc_interval = int(response['msg'].split("interval=")[1].split("SESSION")[0])
```

Comment:
Wouldn't it be safer to use `[0-9]` regexp here? As it's not guaranteed that `SESSION` always follows the interval

Reply:
If I understand correctly, this parsing behavior is based on response format seen in https://gitlab.com/openconnect/openconnect/merge_requests/19#note_122962902

```
{'msg': 'interval=35SESSION=SESSION_POLICIES=timestamp=1544122387', 'connId': '0'}
```

In my tests, a PS server is responding with following content

```
{'timestamp': '<unix timestamp in seconds>', 'interval': '<host checking interval in minutes>', 'SESSION_POLICIES': '', 'SESSION': '', 'connId': '0', 'msg': '<Base 64 encoded text, describing Host Checker policy>'}
```